src/Builder.js

```diff
@@ -24,44 +24,39 @@ class Builder {
 
     const groups = this._buildPolygonGroups(navMesh);
 
-    zone.groups = [];
-
-    const findPolygonIndex = function (group, p) {
-      for (let i = 0; i < group.length; i++) {
-        if (p === group[i]) return i;
-      }
-    };
-
-    // TODO: This block represents 50-60% of navigation mesh construction time,
+    // TODO: This block represents a large portion of navigation mesh construction time
     // and could probably be optimized. For example, construct portals while
     // determining the neighbor graph.
-    groups.forEach((group) => {
+    zone.groups = new Array(groups.length);
+    groups.forEach((group, i) => {
```

Comment:
nit: do you mind spelling out the single-letter variables here and below? for example:

- i --> groupIndex
- p --> polygon
- j --> polygonIndex

This code isn't something I touch very often (mostly I've just refactored and optimized a bit from PatrolJS) so I think the extra readability is helpful.

Reply:
Sure, I'll change it.